example/spec/loadTest/ingestLoad_queueGranPassThruSF.js

```diff
@@ -0,0 +1,253 @@
+        totalGranulesCompleted.push(record);
+      } catch (error) {
+        incompletedGranules.push({ granuleId: granule.granuleId, err: error });
+      }
+    })
+  );
+
+  console.log(greenConsoleLog(),
+    `\n___ ${completedGranules.length}/${inputPayload.granules.length} Granules completed by workflow ${nthWorkflow} ___`);
+
+  if (completedGranules.length !== inputPayload.granules.length) {
+    console.log(redConsoleLog(),
+      `\n___ ${incompletedGranules.length}/${inputPayload.granules.length} Granules incompleted by workflow ${nthWorkflow} ___`);
+
+    incompletedGranules.map((granuleError) =>
+      console.log(redConsoleLog(),
+        `\n${JSON.stringify(granuleError.granuleId)} - ${granuleError.err}`));
+  }
+};
+
+describe('The Granule Ingest Load Test ', () => {
+  beforeAll(async () => {
+    if (!Number.isFinite(Number(granuleCountPerWorkflow)) || granuleCountPerWorkflow > 450) {
+      beforeAllFailed = `===== beforeAll() - Invalid input for number of granules per Workflow detected - ${granuleCountPerWorkflow} =====`;
+      throw new Error(beforeAllFailed);
+    } else {
+      try {
+        config = await loadConfig();
+
+        process.env.GranulesTable = `${config.stackName}-GranulesTable`;
+        process.env.ExecutionsTable = `${config.stackName}-ExecutionsTable`;
+        process.env.CollectionsTable = `${config.stackName}-CollectionsTable`;
+
+        await pMap(
+          new Array(totalWorkflowCount).fill().map((_, index) => index + 1),
+          async (index) => {
+            await batchGranulesProcessing(index);
+          },
+          { concurrency: 8 }
+        );
+      } catch (error) {
+        beforeAllFailed = `===== beforeAll() failed =====\n ${error}`;
+        throw new Error(beforeAllFailed);
+      }
+    }
+  });
+
+  afterAll(async () => {
+    // clean up stack state added by test
+    await Promise.all(
+      totalInputPayloads.map(async (inPayload) =>
+        await cleanupLoadTestGranules(config.stackName, inPayload.granules))
+    );
+
+    console.log(blueConsoleLog(), '\n===== Delete lambdaOutputPayload ExecutionArns =====');
+    queueGranulesChildExecutionArns.forEach((lambdaOutputPayload) => {
+      lambdaOutputPayload.running.forEach((childExecutionArn) => {
+        deleteExecution({
+          prefix: config.stackName,
+          executionArn: childExecutionArn,
+        });
+        // console.log(blueConsoleLog(), `${JSON.stringify(childExecutionArn)}`);
+      });
+    });
+
+    if (queueGranulesExecutionArns !== undefined) {
+      console.log(blueConsoleLog(), '\n===== Delete queueGranules ExecutionArns =====');
+      queueGranulesExecutionArns.forEach((qGranulesExecutionArn) => {
+        deleteExecution({
+          prefix: config.stackName,
+          executionArn: qGranulesExecutionArn,
+        });
+        console.log(blueConsoleLog(), `${JSON.stringify(qGranulesExecutionArn)}`);
+      });
+    }
+
+    await Promise.all(
```

Comment:
Same as above. You _may_ not need a `Promise.all` here. It looks like it would be synchronous. Double-check what's going on in `cleanupProviders` though. I don't think it returns a promise but could be wrong.